derive.go

```diff
@@ -191,12 +194,23 @@ func (mk *MiniSecretKey) HardDeriveMiniSecretKey(i []byte, cc [ChainCodeLength]b
 
 // DeriveKey derives an Extended Key from the Mini Secret Key
 func (mk *MiniSecretKey) DeriveKey(t *merlin.Transcript, cc [ChainCodeLength]byte) (*ExtendedKey, error) {
+	if t == nil {
+		return nil, errors.New("transcript provided is nil")
+	}
+
 	sk := mk.ExpandEd25519()
 	return sk.DeriveKey(t, cc)
 }
 
 func (pk *PublicKey) DeriveKey(t *merlin.Transcript, cc [ChainCodeLength]byte) (*ExtendedKey, error) {
-	sc, dcc := pk.DeriveScalarAndChaincode(t, cc)
+	if t == nil {
```

Comment:
Is this check required? We are already checking this inside `DeriveScalarAndChaincode`